aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -96,20 +100,18 @@ module aptos_framework::block {
         borrow_global<BlockResource>(@aptos_framework).epoch_interval / 1000000
     }
 
-    /// Set the metadata for the current block.
-    /// The runtime always runs this before executing the transactions in a block.
-    fun block_prologue(
-        vm: signer,
+    fun block_prologue_common(
+        account: &signer,
         hash: address,
         epoch: u64,
         round: u64,
         proposer: address,
         failed_proposer_indices: vector<u64>,
         previous_block_votes_bitvec: vector<u8>,
         timestamp: u64
-    ) acquires BlockResource {
+    ): u64 acquires BlockResource {
         // Operational constraint: can only be invoked by the VM.
-        system_addresses::assert_vm(&vm);
+        system_addresses::is_reserved_address(address_of(account));
```

Comment:
Why this change?

Reply:
I was thinking to let vm run this as 0x1 to save some state initialization and check the state with `exists<SomeFlag>()`

But it seems you are suggesting we still run as 0x0 and check the state with something like `global_borrow<SomeFlagWrapper>().boolean_flag`

any difference?